clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -344,13 +344,21 @@ LogicalResult CastOp::verify() {
       return emitOpError() << "requires !cir.int for result";
     return success();
   }
-  case cir::CastKind::int_to_float:
+  case cir::CastKind::int_to_float: {
     if (!srcType.isa<mlir::cir::IntType>())
       return emitOpError() << "requires !cir.int for source";
     if (!resType.isa<mlir::FloatType>())
       return emitOpError() << "requires !cir.float for result";
     return success();
   }
+  case cir::CastKind::bool_to_float: {
+    if (!srcType.isa<mlir::cir::BoolType>())
+      return emitOpError() << "requires !cir.bool for source";
+    if (!resType.isa<mlir::FloatType>())
+      return emitOpError() << " requires !cir.float for result";
+    return success();
```

Comment:
Nice, can you also add testcases for this in `invalid.cir`?

Reply:
Done.